Engine/src/Engine/Scene.cpp

```diff
@@ -25,12 +25,6 @@ namespace Engine
 #pragma region Runtime Functions
 	void Scene::onRuntimeStart()
 	{
-		if (!initializeGL()) return;
```

Comment:
Since initializeGL is private, this change essentially means that only friends of Scene can properly initialize scenes. If deserialization is the only way to initialize new scenes, this should be fine. Either way, this decision should likely be noted in a comment in either in Scene constructor or Scene::initializeGL.

Reply:
I think this is complex enough now that a Renderer class is warranted but outside of the scope of this PR. I made a trello ticket to track this concern: https://trello.com/c/6cMTYYrU/82-create-renderer-class